packages/arb-token-bridge-ui/src/hooks/useAccountIsBlocked.ts

```diff
@@ -0,0 +1,57 @@
+import { useMemo } from 'react'
+import { useAccount } from 'wagmi'
+import useSWRImmutable from 'swr/immutable'
+
+import { ApiResponseSuccess } from '../pages/api/screenings'
+import { trackEvent } from '../util/AnalyticsUtils'
+
+async function isBlocked(address: `0x${string}`): Promise<boolean> {
+  if (
+    process.env.NODE_ENV !== 'production' ||
+    process.env.NEXT_PUBLIC_IS_E2E_TEST
+  ) {
+    return false
+  }
+
+  const searchParams = new URLSearchParams({ address })
+  const response = await fetch('/api/screenings?' + searchParams, {
+    method: 'GET',
+    headers: { 'Content-Type': 'application/json' }
+  })
+
+  if (!response.ok) {
+    return false
+  }
+
+  return ((await response.json()) as ApiResponseSuccess).blocked
+}
+
+async function fetcher(address: `0x${string}`): Promise<boolean> {
+  const accountIsBlocked = await isBlocked(address)
+
+  if (accountIsBlocked) {
+    trackEvent('Address Block', { address })
+  }
+
+  return accountIsBlocked
+}
+
+export function useAccountIsBlocked() {
+  const { address } = useAccount()
+
+  const queryKey = useMemo(() => {
+    if (typeof address === 'undefined') {
+      // Don't fetch
+      return null
+    }
+
+    return ['useAccountIsBlocked', address.toLowerCase()]
+  }, [address])
+
+  const { data: isBlocked } = useSWRImmutable<boolean>(
+    queryKey,
+    ([, _address]) => fetcher(_address)
```

Comment:
nit: You can have the key as `[address.toLowerCase(), 'useAccountIsBlocked']` so you don't need this `_` here anymore

Reply:
> User are now blocked directly when accessing the bridge. On production, it does let you set a transfer amount, and click on "move funds" first.

This isn't true, you're blocked as soon as you connect your wallet. Maybe you just click "connect wallet" a bit later, and in the meantime you're connected with a different account.